packages/fluentui/react-northstar/src/components/Tree/Tree.tsx

```diff
@@ -171,7 +215,61 @@ class Tree extends AutoControlledComponent<WithAsProp<TreeProps>, TreeState> {
     parentRef.current.focus();
   };
 
+  setSelectedItemIds = (e: React.SyntheticEvent, selectedItemIds: string[]) => {
+    _.invoke(this.props, 'onSelectedItemIdsChange', e, { ...this.props, selectedItemIds });
+
+    this.setState({
+      selectedItemIds,
+    });
+  };
+
+  processItemsForSelection = (e: React.SyntheticEvent, treeItemProps: TreeItemProps) => {
+    let { selectedItemIds } = this.state;
+    const { id, selectableParent, items, expanded } = treeItemProps;
+    const treeItemHasSubtree = hasSubtree(treeItemProps);
+    const isExpandedSelectableParent = treeItemHasSubtree && selectableParent && expanded;
+
+    // if the parent is not selectable or is collapsed it means treeItem should be expanded, not procced with selection
+    if ((treeItemHasSubtree && !selectableParent) || (treeItemHasSubtree && !expanded)) {
+      return;
+    }
+
+    // if the target is equal to currentTarget it means treeItem should be collapsed, not procced with selection
+    if (isExpandedSelectableParent && e.target === e.currentTarget) {
+      return;
+    }
+
+    // push all tree items under particular parent into selection array
+    // not parent itself, therefore not procced with selection
+    if (isExpandedSelectableParent) {
+      items.forEach(item => {
```

Comment:
this looks like a `reduce`: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce
 
 ```js
items.reduce((acc, item) => { 
  if (selectedItemIds.indexOf(item['id']) === -1) { 
    acc.push(item['id']) 
  } 
  return acc
}), selectedItemIds);
```

Reply:
Done